core/txpool/ingress_filters.go

```diff
@@ -16,44 +15,34 @@ type IngressFilter interface {
 	FilterTx(ctx context.Context, tx *types.Transaction) bool
 }
 
-type interopFilter struct {
-	logsFn  func(tx *types.Transaction) (logs []*types.Log, logTimestamp uint64, err error)
-	checkFn func(ctx context.Context, ems []interoptypes.Message, safety interoptypes.SafetyLevel, emsTimestamp uint64) error
+type interopFilterAPI interface {
+	CurrentInteropBlockTime() (uint64, error)
+	TxToInteropAccessList(tx *types.Transaction) []common.Hash
+	CheckAccessList(ctx context.Context, inboxEntries []common.Hash, minSafety interoptypes.SafetyLevel, executingTimestamp uint64) error
+}
+type interopAccessFilter struct {
+	api interopFilterAPI
 }
 
-func NewInteropFilter(
-	logsFn func(tx *types.Transaction) ([]*types.Log, uint64, error),
-	checkFn func(ctx context.Context, ems []interoptypes.Message, safety interoptypes.SafetyLevel, emsTimestamp uint64) error) IngressFilter {
-	return &interopFilter{
-		logsFn:  logsFn,
-		checkFn: checkFn,
+func NewInteropFilter(api interopFilterAPI) IngressFilter {
+	return &interopAccessFilter{
+		api: api,
 	}
 }
 
 // FilterTx implements IngressFilter.FilterTx
-// it gets logs checks for message safety based on the function provided
-func (f *interopFilter) FilterTx(ctx context.Context, tx *types.Transaction) bool {
-	logs, logTimestamp, err := f.logsFn(tx)
+// it uses provided functions to get the access list from the transaction
+// and check it against the supervisor
+func (f *interopAccessFilter) FilterTx(ctx context.Context, tx *types.Transaction) bool {
+	// if CurrentInteropBlockTime returns an error, we assume that the chain is not set up for interop
+	// in which case we allow all transactions
+	time, err := f.api.CurrentInteropBlockTime()
```

Comment:
We don't want interop txs to bypass the ingress filter just before interop activates.
Before interop activates I think we may want to just always drop txs that have an access-list that contains the cross-l2-inbox address.